src/infra/marketplace/marketplaceCmdRepo.go

```diff
@@ -54,9 +56,14 @@ func (repo *MarketplaceCmdRepo) installServices(
 			shouldCreatePhpVirtualHost = true
 		}
 
-		_, err := servicesQueryRepo.ReadByName(serviceWithVersion.Name)
-		if err == nil {
-			continue
+		shouldIncludeMetrics := false
+		readInstalledServiceDto := dto.ReadInstalledServicesItemsRequest{
+			ServiceName:          &serviceWithVersion.Name,
+			ShouldIncludeMetrics: &shouldIncludeMetrics,
+		}
+		_, err := servicesQueryRepo.ReadOneInstalledItem(readInstalledServiceDto)
+		if err != nil && err.Error() != "ServiceInstalledItemNotFound" {
+			return err
 		}
```

Comment:
That's weird. Wasn't ReadOne suppose to return err if it does not find one? Telling you that the nomenclature is important.

Reply:
By the way, the chances of a error string changing is high. If you really need to check if an error is X, then convert the error into an enum/const and check for the const. That way if the error ever changes on the const, it will also changes where you check for it. Never check an error by string if you can avoid. Even if you can't, you'll most likely be better served with a regex match.